R/chromote_session.R

```diff
@@ -535,19 +538,36 @@ ChromoteSession <- R6Class(
       private$event_manager$invoke_event_callbacks(event, params)
     },
 
-    #' @description
-    #' Disable callbacks for a given session.
-    #'
-    #' For internal use only.
-    mark_closed = function() {
-      private$is_active_ <- FALSE
+    #' @description Mark a session, and optionally, the underlying target,
+    #'   as closed. For internal use only.
+    #' @param target_closed Has the underlying target been closed as well as the
+    #'   active debugging session?
+    mark_closed = function(target_closed) {
+      private$session_is_active <- FALSE
+      private$target_is_active <- FALSE
```

Comment:
It looks like `target_closed` is unused?

Reply:
Doh. I realised that I forgot to call `spawn()` before closing in my script test. Here's the new output:

```R
library(chromote)

sess <- ChromoteSession$new()
sess$check_active()
#> NULL

# SLEEP -----------------------

sess$check_active()
#> Error in `sess$check_active()`:
#> ! Session has been closed.
#> ℹ Call session$respawn() to create a new session that connects to the same target.

sess2 <- sess$respawn()
#> ! Reconnecting to chrome process.
#> ℹ All active sessions will be need to be respawned.

sess$close()

sess$respawn()
#> Error in `sess$respawn()`:
#> ! Can't respawn session; target has been closed.
```